runtime-testsuite/test/org/antlr/v4/test/runtime/csharp/BaseCSharpTest.java

```diff
@@ -245,8 +245,7 @@ public String execLexer(String grammarFileName,
 	Set<String> sourceFiles = new HashSet<String>();
 
 	private void addSourceFiles(String ... files) {
-		for(String file : files)
-			this.sourceFiles.add(file);
+		this.sourceFiles.addAll(Arrays.asList(files));
```

Comment:
There might be better optimization here:
```
Collections.addAll(this.sourceFiles, files);
```
As per Javadoc, its function remains the same, but it is likely to run significantly faster under most implementations

Reply:
@NathanQingyangXu 
Hi.
After further test I notice you are right, this is a wrong optimize, it runs slightly slower in HashSet implementation.
Should revoke to original.